service/common/src/main/java/org/apache/polaris/service/events/BeforeRequestRateLimitedEvent.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.service.events;
+
+/**
+ * Emitted before the RateLimiterFilter rejects a request due to exceeding the rate limit.
+ *
+ * @param method The request's HTTP method
+ * @param absolutePath The request's absolute path
+ */
+public record BeforeRequestRateLimitedEvent(String method, String absolutePath)
```

Comment:
What's the use case of this one and how would an implementation behave in case of a DoS attack attempt?

Reply:
See https://github.com/apache/polaris/pull/922#discussion_r2001532609

Can you elaborate on your DoS concerns? If you're using the no-op listener, nothing changes. It's up to you to respond to events in your preferred way.

Reply:
I'd like to counter the question: What's the _concrete_ use case for this one?

Reply:
Logging.

Reply:
People have different logging requirements. Logs are a controversial change, which you explained elegantly [here](https://github.com/apache/polaris/pull/458#pullrequestreview-2451856115).

Reply:
Sorry, I'm strongly -1 on this event. DoS w/ millions or billions of requests are already an issue - this event just makes such a situation worse.

Reply:
That's the only one that's there - "noop". The code yields exactly nothing. The events can't be consumed by anybody. I would really like to see concrete use cases for Apache Polaris users, how the architecture would look like and how things are integrated.

The missing attributes that I noticed let me think that the consuming side (aka use cases) hasn't been thought through.

Reply:
> The code yields exactly nothing. The events can't be consumed by anybody.

Isn't exactly the opposite true? After this PR, events can be consumed by everybody! 

I've listed demonstrable use cases for each event in the design doc.